torch/nn/_functions/thnn/auto.py

```diff
@@ -40,7 +40,7 @@ def symbolic(*args, **kwargs):
 
     @staticmethod
     def forward(ctx, input, target, *args):
-        ctx._backend = type2backend[type(input)]
+        ctx._backend = type2backend[input.type()]
```

Comment:
We shouldn't be using strings here... `.type()` is super slow.

Reply:
It's ~650ns.

The number of functions going through this code path is shrinking. It's just RNNs and a few functions we haven't ported to ATen yet.